### ISSUE TITLE
get_version improvements

### ISSUE THREAD

**Issue by benjaoming** [learningequality/kolibri#1486]
## Summary

From today's dev build pipeline meeting..

To not be dependent on neither git data nor a VERSION file, this can be done:

 * Create a `VERSION` tuple once more
 * ~Remove VERSION file stuff~
 * Keep existing functionality for reading git logs and tags into versioning
 * Create a clear warning in case of inconsistencies between the tuple data and the git tag

Also dropping this note: Need to investigate *if* in fact it is guaranteed to be a Kolibri git repo, in order to avoid some inconsistent inconsistency warning.

## System information

Targeting current `develop` branch. There's no urgency about getting this merged. (expecting that it will work without breakage, though)

## Background

I'm using PEP-440: https://www.python.org/dev/peps/pep-0440/

Here's the example about development versions

> X.YaN.devM       # Developmental release of an alpha release
> X.YbN.devM       # Developmental release of a beta release
> X.YrcN.devM      # Developmental release of a release candidate
> X.Y.postN.devM   # Developmental release of a post-release

## Auto-added version hashes, semi-auto, and no hashes (`.dev123`)

Auto-applied version hashing:

We assume there is no git tagging here, we just apply git commit hashes or timestamps.

 * `(0, 1, 0, 'alpha', 0)` - the 0th version of an alpha is by definition **unreleased** and ALWAYS converted to `0.1.dev1234`. This is because nothing comes before a0 anyways.. and developing for the first alpha release just means doing a develop version. This will always rank lower than any succeeding alpha releases.
 * `(0, 1, 0, 'beta', 0)` - the 0th version of a beta is by definition **unreleased**.... but `0.1b123` would rank higher than actually releasing `0.1b1`! So we have to apply PEP-440's suggestion of post-fixing a dev-thing, like this: `0.1b0.dev123`. Of course, this is a bit weird, so it's better to avoid `(0, 1, 0, 'beta', 0)` because developing on the 0th beta doesn't make sense... we would always develop for the 1st beta. So essentially, this version tuple should IMO raise an exception.
 * `(0, 1, 0, 'rc', 0)` - see above. Could be a candidate for fully-automatic tagging, but should IMO raise an exception.

Semi-automatic release hashing w/ tag info:

If the current commit doesn't match the git tag, then we postfix with `.dev123`. If there is no git data, we assume the `VERSION` tuple as is.

 * `(0, 4, 0, 'alpha', 1+)` - the 1st version of an alpha is by definition a development release, and it makes sense to check what git meta data, we have available. 
 * `(0, 4, 0, 'beta', 1+)` -  the 1st version of a beta is also something for development, so we also add Git data.
 * `(0, 1, 0, 'rc', 1+)` - same as above.

No release hashing:

It's possible to do a consistency check on the git tag, but IMO this is better done during build time. Otherwise we'll have subprocess calls to `git` everytime `kolibri` is running in a production system, which is a bad idea IMO.

 * `(0, 4, 0, 'final', 0)` -  A 0th final release just means a final release. **Nothing comes in between a release candidate and a final release.**
 * `(0, 4, 0, 'final', 1)` -  This is a post release and becomes `0.4.post1`

## VERSION file

It seems necessary to keep the version file.

In the case a development release `(0, 1, 0, 'alpha', 0)` = `0.1.dev123` is released without the git information cause `123`, the version would change back to `0.1.dev0` when code is loaded outside of the git repo. So @aronasorman's initial idea was spot on.

Version files are only relevant in cases where there's auto-generated data in the version.

## Real-life consequences (anything community should be aware of, for instance how it affects your deployment)

Stuff like #1077

**Comment by benjaoming** [comment]
This looks nice:

```
version = (0, 1, 0, 'alpha', 1), git_version = (0, 4, 0, 'beta1', '28', 'g15e8bee')

    def assert_git_version(version, git_version):
        if not (
            version[0] == git_version[0] and
            version[1] == git_version[1] and
            version[2] == git_version[2] and
            version[3] == git_version[3]
        ):
            raise AssertionError(
>               "Inconsistent git tagging: {} != {}".format(version, git_version)
            )
E           AssertionError: Inconsistent git tagging: (0, 1, 0, 'alpha', 1) != (0, 4, 0, 'beta1', '28', 'g15e8bee')
```

**Comment by benjaoming** [comment]
Reading through this issue, I would say that the target has been met in #1498 and we can work from there. The description in this issue and the PR summary is all clearly reflected in new documentation.

**Comment by rtibbles** [comment]
Fixed in #1498
